### PR TITLE
Added support for DivIcon

### PR DESCRIPTION
**Pull request by omanges**
Added support for DivIcon, related issue https://github.com/jupyter-widgets/ipyleaflet/issues/246

### CODE REVIEW

**Comment by martinRenou** [comment]
Neat!!

File: docs/source/api_reference/divicon.rst [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+----------
+
+=====================    =====================   ===
+Attribute                Default Value           Doc
+=====================    =====================   ===
+html                     ''                      Custom HTML code to put inside the div element,
+bg_pos                   (0, 0)                  Optional relative position of the background, in pixels.
+icon_size                (10, 10)                size icon will be rendered
+icon_anchor              (0, 0)                  anchor point of icon
+class_name               leaflet-div-icon        A custom class name to assign to icon.
```
Comment (martinRenou) [inline docs/source/api_reference/divicon.rst:31]: I know this is here in some classes already, but I actually think we should not expose any `class_name` property to Python.

The reason is that users don't have a non-hacky way to inject CSS into the page, so I think we should not rely on this property.

File: ipyleaflet/leaflet.py [hunk 2]
```diff
@@ -283,6 +283,56 @@ def _validate_attr(self, proposal):
 
         return value
 
+class DivIcon(UILayer):
+    """DivIcon class.
+
+    Custom lightweight icon for markers that uses a simple <div> element 
+    instead of an image used for markers.
```
Comment (martinRenou) [inline ipyleaflet/leaflet.py:290]: Thanks a ton for this!

**Review by martinRenou — COMMENTED** [review]
I just have a single comment, apart from that thanks very much for this PR!

File: docs/source/api_reference/divicon.rst (hunk 1 above)
Comment (omanges) [reply docs/source/api_reference/divicon.rst:31]: Yes right makes sense.

**Comment by martinRenou** [comment]
Thanks!

**Comment by omanges** [comment]
It seems I have now a very good understanding on ipyleaflet 😬

**Pull request merged**